packages/vow/src/tools.js

```diff
@@ -22,6 +22,18 @@ export const prepareVowTools = (zone, powers = {}) => {
   const makeWatchUtils = prepareWatchUtils(zone, watch, makeVowKit);
   const watchUtils = makeWatchUtils();
   const asVow = makeAsVow(makeVowKit);
+  /**
+   * TODO FIXME make this real
+   * Create a function that retries the given function if the underlying
+   * functions rejects due to upgrade disconnection.
+   *
+   * @param {string} name
+   * @param {(...args: unknown[]) => unknown} fn
+   */
+  const retriable =
+    (name, fn) =>
+    (...args) =>
+      watch(fn(...args));
```

Comment:
As an endowment this could work, since the visible endowments do not need to be durable. Instead, they must be re-endowed on prepare in each incarnation.